src/python/WMCore/MicroService/MSPileup/MSPileup.py

```diff
@@ -4,14 +4,125 @@
 Description: MSPileup provides logic behind the pileup WMCore module.
 """
 
+# system modules
+from threading import current_thread
+
+# 3rd party modules
+import cherrypy
+
 # WMCore modules
+from WMCore.REST.Auth import authz_match
 from WMCore.MicroService.MSCore.MSCore import MSCore
+from WMCore.MicroService.DataStructs.DefaultStructs import PILEUP_REPORT
+from WMCore.MicroService.MSPileup.MSPileupData import MSPileupData
+
+
+def mspileupError(doc):
```

Comment:
Valentin, can you please provide a unit test for this function?

Reply:
done